ddtrace/llmobs/_integrations/openai.py

```diff
@@ -199,20 +199,24 @@ def _llmobs_set_meta_tags_from_chat(
             return
         output_messages = []
         for idx, choice in enumerate(resp.choices):
+            tool_calls_info = []
             content = getattr(choice.message, "content", "")
             if getattr(choice.message, "function_call", None):
-                content = "[function: {}]\n\n{}".format(
-                    getattr(choice.message.function_call, "name", ""),
-                    getattr(choice.message.function_call, "arguments", ""),
-                )
+                function_call_info = {
+                    "name": getattr(choice.message.function_call, "name", ""),
+                    "arguments": json.loads(getattr(choice.message.function_call, "arguments", {})),
+                }
+                content = {"function_call": function_call_info}
             elif getattr(choice.message, "tool_calls", None):
-                content = ""
                 for tool_call in choice.message.tool_calls:
-                    content += "\n[tool: {}]\n\n{}\n".format(
-                        getattr(tool_call.function, "name", ""),
-                        getattr(tool_call.function, "arguments", ""),
-                    )
-            output_messages.append({"content": str(content).strip(), "role": choice.message.role})
+                    tool_call_info = {
+                        "name": getattr(tool_call.function, "name", ""),
+                        "arguments": json.loads(getattr(tool_call.function, "arguments", {})),
+                        "tool_id": getattr(tool_call, "id", ""),
+                        "type": getattr(tool_call, "type", ""),
+                    }
+                    tool_calls_info.append(tool_call_info)
+            output_messages.append({"content": "", "role": choice.message.role, "tool_calls": tool_calls_info})
```

Comment:
In case an openai response contains both content and tool calls, can we ensure that we capture both in the same message? Right now it looks like we set content to be an empty string.

Reply:
yeah sorry about that - i meant to set it to just content which will default to an empty string if it's empty and still store content if both content and tool calls are present